Relax parsing of labels

mbartlett21 opened the pull request:
Let it accept more commands

Mark-Simulacrum commented on parser/src/command.rs:
```diff
@@ -242,7 +242,7 @@ fn edit_1() {
 #[test]
 fn edit_2() {
-    let input_old = "@bot modify label: +bug.";
+    let input_old = "@bot modify labeled: +bug.";
```
Hm, I'm not entirely sure I'm following this test (before or modified), but labeled doesn't look right to me, since I don't think that parses at all right now? We probably want to write a comment here and update the test so that it has the same behavior in terms of what things parse to (and maybe assert that as well).

mbartlett21 replied:
@Mark-Simulacrum It is testing an error there with the assert_ne! below.

https://github.com/rust-lang/triagebot/pull/1591/files#diff-1eb763ec24cd0509bade43f86001d94744947818b6ea7596c4e003c858327395R249

Mark-Simulacrum replied:
There's no guarantee as to what will actually happen in that assert, though? It could just as easily be two different parses or something like that, which seems not great.

mbartlett21 replied:
Agreed. I'll take a look at rewriting a new test

Mark-Simulacrum commented:
Sorry this slipped by my radar for so long -- I think it looks good overall, just one small comment.

mbartlett21 commented:
@Mark-Simulacrum 
I've simplified the parsing (and allowed using `@rustbot labels to: +this -that`), and updated the tests to test various inputs.

@rustbot ready

mbartlett21 commented:
@Mark-Simulacrum 

These are the tests that are failing now, since I allowed `@bot labels to: +this`. Should these still provide a `MisleadingTo` error?

https://github.com/rust-lang/triagebot/blob/83f7a8df8483954e8e6e59feb1a80b1f111c228c/parser/src/command/relabel.rs#L270-L292

Mark-Simulacrum commented:
Probably that shouldn't be an error anymore.

mbartlett21 commented:
@Mark-Simulacrum 
Ok. This is ok to merge now.
